weave/trace/op.py

```diff
@@ -407,37 +414,52 @@ def _do_call(
     if not pargs:
         pargs = _default_on_input_handler(op, args, kwargs)
 
+    # Handle all of the possible cases where we would skip tracing.
     if settings.should_disable_weave():
         res = func(*pargs.args, **pargs.kwargs)
-    elif weave_client_context.get_weave_client() is None:
+        return res, call
+    if weave_client_context.get_weave_client() is None:
+        res = func(*pargs.args, **pargs.kwargs)
+        return res, call
+    if not op._tracing_enabled:
+        res = func(*pargs.args, **pargs.kwargs)
+        return res, call
+    if not get_tracing_enabled():
         res = func(*pargs.args, **pargs.kwargs)
-    elif not op._tracing_enabled:
+        return res, call
+
+    current_call = call_context.get_current_call()
+    if current_call is None:
```

Comment:
might be nice to say: if your parent is traced, then you are (as a comment)

Reply:
Note: this specific case is not tested by your unit tests (setting a child sample rate to 0, and a parent to 1)

Reply:
Good point